test/qbwc/integration/response_test.rb

```diff
@@ -104,57 +168,118 @@ def handle_response(resp, job, request, data)
+    QBWC.add_job(:multiple_request_job, true, COMPANY, MultiRequestWorker)
+    _test_warning_then_error(0)
+  end
 
-    # Simulate controller receive_response
-    session.response = QBWC_CUSTOMER_QUERY_RESPONSE_ERROR
-    assert_equal '3120',  session.status_code
-    assert_equal 'Error', session.status_severity
-    assert_equal "QBWC ERROR: 3120 - #{QBWC_CUSTOMER_QUERY_STATUS_MESSAGE_ERROR}", session.error
+  test "processes warning then error stop 2requests byargument" do
+    QBWC.on_error = :stop
+    QBWC.add_job(:multiple_request_job, true, COMPANY, HandleResponseWithDataWorker, [QBWC_CUSTOMER_QUERY_RQ, QBWC_CUSTOMER_QUERY_RQ])
+    _test_warning_then_error(0)
   end
 
-  test "processes error then warning" do
+  test "processes warning then error continue 2requests byargument" do
     QBWC.on_error = :continue
+    QBWC.add_job(:multiple_request_job, true, COMPANY, HandleResponseWithDataWorker, [QBWC_CUSTOMER_QUERY_RQ, QBWC_CUSTOMER_QUERY_RQ])
+    _test_warning_then_error(0)
+  end
 
-    # Add jobs
+  test "processes error then warning stop 2jobs" do
+    QBWC.on_error = :stop
     QBWC.add_job(:query_joe_customer,       true, COMPANY, HandleResponseWithDataWorker)
     QBWC.add_job(:query_joe_customer_again, true, COMPANY, HandleResponseWithDataWorker)
+    _test_error_then_warning_that_stops
+  end
 
-    # Simulate controller authenticate
-    ticket_string = QBWC::ActiveRecord::Session.new(QBWC_USERNAME, COMPANY).ticket
-    session = QBWC::Session.new(nil, COMPANY)
+  test "processes error then warning continue 2jobs" do
+    QBWC.on_error = :continue
+    QBWC.add_job(:query_joe_customer,       true, COMPANY, HandleResponseWithDataWorker)
+    QBWC.add_job(:query_joe_customer_again, true, COMPANY, HandleResponseWithDataWorker)
+    _test_error_then_warning
+  end
 
-    # Simulate controller receive_response
-    session.response = QBWC_CUSTOMER_QUERY_RESPONSE_ERROR
-    assert_equal '3120',  session.status_code
-    assert_equal 'Error', session.status_severity
-    assert_equal "QBWC ERROR: 3120 - #{QBWC_CUSTOMER_QUERY_STATUS_MESSAGE_ERROR}", session.error
+  test "processes error then warning stop 2requests byworker" do
+    QBWC.on_error = :stop
+    QBWC.add_job(:multiple_request_job, true, COMPANY, MultiRequestWorker)
+    _test_error_then_warning_that_stops
+  end
 
-    # Simulate controller send_request
-    assert_not_nil session.next_request
+  test "processes error then warning continue 2requests byworker" do
+    QBWC.on_error = :continue
+    QBWC.add_job(:multiple_request_job, true, COMPANY, MultiRequestWorker)
+    _test_error_then_warning(0)
+  end
 
-    # Simulate controller receive_response
-    session.response = QBWC_CUSTOMER_QUERY_RESPONSE_WARN
-    assert_equal '500',  session.status_code
-    assert_equal 'Warn', session.status_severity
-    assert_equal "QBWC WARN: 500 - #{QBWC_CUSTOMER_QUERY_STATUS_MESSAGE_WARN}", session.error
+  test "processes error then warning stop 2requests byargument" do
```

Comment:
Maybe I'm missing something, but how does this test check to make sure the second request doesn't run?